lib/dynamoid/criteria/chain.rb

```diff
@@ -178,8 +178,8 @@ def pages_via_query
       def pages_via_scan
         return enum_for(:pages_via_scan) unless block_given?
 
-        Dynamoid.adapter.scan(source.table_name, scan_query, scan_opts).each do |items|
-          yield items.map { |hash| source.from_database(hash) }
+        Dynamoid.adapter.scan(source.table_name, scan_query, scan_opts).each do |items, metadata|
+          yield items.map { |hash| source.from_database(hash) }, metadata
```

Comment:
I would emphasises here we path only one particular key in `metadata` with `metadata.slice(: last_evaluated_key)`

Reply:
Sure thing